src/Context/FilterContext/FilterUtils.js

```diff
@@ -0,0 +1,60 @@
+const composeFunction = (state, functionList, data) =>
+  functionList.reduce((accumulator, currentFunction) => {
+    return currentFunction(accumulator, state);
+  }, data);
+
+const productListByRange = (productList, state) =>
+  state.sortByRange === 0
+    ? productList
+    : [...productList].filter((product) => product.price <= state.sortByRange);
```

Comment:
suggestion: filter doesn't mutate the original array so there no need to apply filter on a copied array, you can directly apply to original array too